launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -455,6 +555,26 @@ private ProcessBuilder createBuilder() {
     for (Map.Entry<String, String> e : builder.childEnv.entrySet()) {
       pb.environment().put(e.getKey(), e.getValue());
     }
+
+    if (workingDir != null) {
+      pb.directory(workingDir);
+    }
+
+    // Only one of redirectError and redirectError(...) can be specified
+    if (redirectErrorStream && errorStream != null) {
```

Comment:
minor: you can use `checkState` (from `CommandBuilderUtils` which is already imported).